extensions/podman/src/podman-cli.ts

```diff
@@ -16,25 +16,36 @@
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
 import { spawn } from 'node:child_process';
-
-import type * as extensionApi from '@tmpwip/extension-api';
+import * as extensionApi from '@tmpwip/extension-api';
 import { isMac, isWindows } from './util';
 
 const macosExtraPath = '/usr/local/bin:/opt/homebrew/bin:/opt/local/bin:/opt/podman/bin';
 
 export function getInstallationPath(): string {
   const env = process.env;
+  let path: string = env.Path;
+
+  // Get the path from the configuration if it's been set
+  const containerConfiguration = extensionApi.configuration.getConfiguration('podman');
+  const userPath = containerConfiguration.get('machine.binary');
+
+  // Determine the correct path to use if Windows or Mac
   if (isWindows) {
-    return `c:\\Program Files\\RedHat\\Podman;${env.PATH}`;
+    path = `c:\\Program Files\\RedHat\\Podman;${env.PATH}`;
   } else if (isMac) {
     if (!env.PATH) {
-      return macosExtraPath;
+      path = macosExtraPath;
     } else {
-      return env.PATH.concat(':').concat(macosExtraPath);
+      path = env.PATH.concat(':').concat(macosExtraPath);
     }
-  } else {
-    return env.PATH;
   }
+
+  // If userPath is not undefined, append to path
+  if (userPath) {
+    path = path.concat(':').concat(userPath.toString());
+  }
+
+  return path;
 }
 
 export function getPodmanCli(): string {
```

Comment:
shouldn't it be in getPodmanCli() that we should use the configuration ?

let say I use the name as configuration '/tmp/my-custom-podman' I expect to have the 'podman machine init' command being instead calling /tmp/my-custom-podman machine init'

Here it'll still search for a podman binary while it seems we get requests to change also the name of the executable

Reply:
Yup, that's intentional as I interpreted the issue as adding an additional $PATH, not overriding the default set one.

We could instead implement https://github.com/containers/podman-desktop/pull/941/files#r1036245642 after we get an update from: https://github.com/containers/podman-desktop/issues/810